receiver/elasticsearchreceiver/metadata.yaml

```diff
@@ -483,9 +499,228 @@ metrics:
   elasticsearch.os.memory:
     description: Amount of physical memory.
     unit: By
+    gauge:
+      value_type: int
+    attributes: [memory_state]
+    enabled: true
+  elasticsearch.indexing_pressure.memory.coordinating:
+    description: Memory consumed, in bytes, by indexing requests in the coordinating stage.
+    unit: By
+    gauge:
+      value_type: int
+    attributes: []
+    enabled: true
+  elasticsearch.indexing_pressure.memory.primary:
+    description: Memory consumed, in bytes, by indexing requests in the primary stage.
+    unit: By
+    gauge:
+      value_type: int
+    attributes: []
+    enabled: true
+  elasticsearch.indexing_pressure.memory.replica:
```

Comment:
I honestly don't know if they should be combined into a single metric. I prefer to keep it separated since 1) it is easier to determine exactly what metrics are missing when comparing the receiver documentation with https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-stats.html and 2) it represents exactly the data structure returned by the source. I'll combine them since there is a preference by you to do so.

However, I would like to know if there is a long-term spec in the planning to govern metric formation. I believe there was a recent PR to have separate out metrics with a 'direction' attribute, which is not backwards compatible.

Reply:
Thanks. It is generally intended with the OTel data model that data points be combined when they can be summed to a _meaningful_ total.

Reply:
Understood. Thanks!